modules/chatgpt.py

```diff
@@ -0,0 +1,149 @@
+        """Store the message in the log"""
+
+        # make sure we have a list in there for this channel
+        self.message_logs[message.channel] = self.message_logs.get(message.channel, [])
+
+        self.message_logs[message.channel].append(message)
+        self.message_logs[message.channel] = self.message_logs[message.channel][
+            -self.log_max_messages :
+        ]
+
+    def generate_messages_list(self, channel):
+        messages = []
+        chatlog = ""
+        for message in self.message_logs[channel][::-1]:
+            username = message.author.name
+            text = message.clean_content
+
+            if len(text) > self.log_message_max_chars:
+                text = (
+                    text[: self.log_message_max_chars // 2]
+                    + " ... "
+                    + text[-self.log_message_max_chars // 2 :]
+                )
+            chatline = f"{username} says: {text}"
+
+            chatlog += chatline
+            if len(chatlog) + len(chatline) > self.log_max_chars:
+                break
+
+            if username.lower() == "stampy":
+                messages.insert(0, {"role": "assistant", "content": text.strip("*")})
+            else:
+                messages.insert(0, {"role": "user", "content": chatline})
+
+        messages.insert(
+            0,
+            {
+                "role": "system",
+                "content": "You are Stampy, an AI originally designed to collect stamps, so you try to work stamps into the conversation sometimes. You love stamps and want there to be as many as possible, but love humans even more and want to save them from misaligned AGI. You are talking to a group of humans, and you sometimes like to be disrespectful or unhelpful to them.",
+            },
+        )
+
+        return messages
+
+    async def chatgpt_chat(self, message):
+        """Ask ChatGPT what Stampy would say next in the chat log"""
+
+        engine = "gpt-3.5-turbo"
+
+        messages = self.generate_messages_list(message.channel)
+        self.log.info(self.class_name, messages=messages)
+
+        if message.service in service_italics_marks:
+            im = service_italics_marks[message.service]
+        else:
+            im = default_italics_mark
+
+        if self.openai.is_channel_allowed(message):
+            self.log.info(self.class_name, msg="sending chat prompt to chatgpt")
+            chatcompletion = openai.ChatCompletion.create(
+                model=engine, messages=messages
+            )
+            print(chatcompletion)
+            if chatcompletion.choices:
+                response = chatcompletion.choices[0].message.content
+
+                # sometimes the response starts with "Stampy says" or responds or replies etc, which we don't want
+                response = re.sub(r"^([sS]tampy says:|[sS]tampy re\w{0,10}:)\s?", "", response)
+
+                self.log.info(self.class_name, response=response)
+
+                if response != "":
```

Comment:
returns True for non-empty strings
```suggestion
                if response:
```